File: guides/micronaut-jms-aws-sqs/java/src/main/java/example/micronaut/DemoProducer.java
package example.micronaut;

import io.micronaut.jms.annotations.JMSProducer;
import io.micronaut.jms.annotations.Queue;
import io.micronaut.messaging.annotation.MessageBody;

import static io.micronaut.jms.sqs.configuration.SqsConfiguration.CONNECTION_FACTORY_BEAN_NAME;

@JMSProducer(CONNECTION_FACTORY_BEAN_NAME) // <1>
public interface DemoProducer {

    @Queue("demo_queue")   // <2>
    void send(@MessageBody String body);  // <3>
}

File: guides/micronaut-jms-aws-sqs/kotlin/src/main/kotlin/example/micronaut/DemoProducer.kt
package example.micronaut

import io.micronaut.jms.annotations.JMSProducer
import io.micronaut.jms.annotations.Queue
import io.micronaut.jms.sqs.configuration.SqsConfiguration.CONNECTION_FACTORY_BEAN_NAME
import io.micronaut.messaging.annotation.MessageBody

@JMSProducer(CONNECTION_FACTORY_BEAN_NAME)  // <1>
interface DemoProducer {

    @Queue("demo_queue")  // <2>
    fun send(@MessageBody body: String?)  // <3>
}